### ISSUE TITLE
Possible to connect external ReactJS website to Odoo with standard hosting via GraphQL?

### ISSUE THREAD

**Issue by timcouchoud** [divad1196/odoo_graphql#5]
Hi,
Thanks for developping this module. 

Do you know if it is technically possible  to connect an Odoo database, hosted in the "standard" cloud (NOT odoo.sh, so without dev possibilities), with an external ReactJS app.

Thanks.

**Comment by divad1196** [comment]
Hi @timcouchoud 
I think you misunderstanding something really important:
![image](https://user-images.githubusercontent.com/43037556/182029773-55bc6f93-4050-4dfc-9169-a1fc3bc49bb1.png)
This module is just a tool to retrieve data in an easier, faster and cleaner way than what is currently possible using jsonrpc and existing API.

## Context
* Odoo already have an API
* Odoo provides a generic API through jsonrpc calls (if you don't know about it, I strongly recommend you to check it)
  I am not aware of an existing library for simplifying jsonrpc calls in javascript, but this is quite easy to implement

## The issue
So what prevents you from using this? The response is: CORS (Cross Origin Resource Sharing)
Odoo "does not allow it". In fact, Odoo is never told to tell the browser to accept a cors-request => the browser won't let your site query Odoo.

## What can you do?
* Odoo lets you define your cors **per route** which is not ideal. Especially, you would need to rewrite almost every routes to specify the hosts (port and protocol) that must be able to query the routes when you want your website to be granted the exact same permissions as if it was the "same-site policy"
* Add a proxy in front of your Odoo to rewrite all responses for a given website to allow the CORS (This is the way to go)
  This has 2 downsides:
  * If you already have a proxy, e.g. On Odoo online there is a proxy which is managing the httpS encryption, you will have a second layer which is not ideal but not that serious, but since the proxy must be able to overwrite the request, it would need to:
    1. Decrypt the response
    2. Rewrite what is needed
    3. Rewrite it again
    
    Which means you will have an decryption/encryption layer in addition which add an overhead
* You will need to host your own proxy at somepoint

## TLDR;
Just set up an nginx proxy somewhere by following [this](https://github.com/divad1196/odoo_nginx_proxy) configuration/tutorial
